### PR TITLE
Revise CLI Documentation

### PR DESCRIPTION
**Pull request by c-bebop**
This PR addresses issue #56 

Most of the changes are minimal, and only represent revisions of already present documentation. Only the `queue` documentation is new.

The rendering of the documentation is available under: https://simexpal-fork.readthedocs.io/en/simexpal_doc_56/command_line_reference.html

### CODE REVIEW

File: docs/command_line_reference.rst [hunk 1]
```diff
@@ -41,68 +44,99 @@ command above).
...
+:--configure:     Configures the build.
+:--delete-source: Deletes the source directory when purging.
+:--install:       Installs the build files.
+:--purge:         Deletes all related build files.
+:--recheckout:    Deletes the cloned git repository, re-clones, regenerates,
+                  reconfigures, re-compiles, and reinstalls it.
+:--recompile:     Re-compiles and reinstall the build.
+:--reconfigure:   Re-configures the build.
+:--regenerate:    Re-generates the build.
+:--reinstall:     Re-installs the build.
```
Comment (fabratu) [inline docs/command_line_reference.rst:58]: We should be consistent concerning the prefix "re". According to the Oxford Dictionary, it appears that writing all in one word is the correct scheme. Of course, there are some words, which the dictionary doesn't contain.

File: docs/command_line_reference.rst [hunk 2]
```diff
@@ -41,68 +44,99 @@ command above).
...
-:launch: launches all the non executed experiments.
-:print: displays all experimental output, including error outputs, on the command-line.
-:purge: deletes the experimental data. To confirm this action it needs the ``-f`` argument.
-:kill: terminates jobs submitted to or started by Slurm. To confirm this action it needs
-    the ``-f`` argument.
+Responsible to check, execute and remove experiments. The experiments command
+accepts the following actions flags:
+
+:info:   Displays all related instances, instance sets, variant axes and variants
+         of experiments on the command-line.
```
Comment (fabratu) [inline docs/command_line_reference.rst:71]: Remove the hyphen in `command-line`.

File: docs/command_line_reference.rst [hunk 3]
```diff
@@ -41,68 +44,99 @@ command above).
...
+
+:list:   Lists all the experiments. Executed experiments are shown in green,
+         failed ones are shown in red, running ones in yellow, and the non
+         executed ones in the default command-line color. With the argument
```
Comment (fabratu) [inline docs/command_line_reference.rst:75]: Same here.

File: docs/command_line_reference.rst [hunk 4]
```diff
@@ -41,68 +44,99 @@ command above).
...
+         ``--detailed`` it will show every single run. With ``--compact`` all
```
Comment (fabratu) [inline docs/command_line_reference.rst:76]: With ``--compact``,

File: docs/command_line_reference.rst [hunk 5]
```diff
@@ -41,68 +44,99 @@ command above).
...
+         runs with the same experiment will be grouped together. The ``--full``
+         option forces simexpal to display the full experiment name.
+
+:launch: Launches all the non executed experiments.
+
+:print:  Displays all experimental output, including error outputs, on the command-line.
```
Comment (fabratu) [inline docs/command_line_reference.rst:82]: Same here.

File: docs/command_line_reference.rst [hunk 6]
```diff
@@ -41,68 +44,99 @@ command above).
...
+
+:purge:  Deletes the experimental data. To confirm this action it needs the ``-f`` argument.
+
+:kill:   Terminates jobs submitted to or started by Slurm. To confirm this
```
Comment (fabratu) [inline docs/command_line_reference.rst:86]: What happens if the command is used with other launchers than Slurm? I would presume these jobs would also get killed, right?

File: docs/command_line_reference.rst [hunk 7]
```diff
@@ -41,68 +44,99 @@ command above).
...
-:``--unfinished``: selects all the unfinished experiments.
-:``--revision <i>``: selects all experiments with the revision named ``r``.
-:``--variants [variants...]``: selects all experiments with the variants from the space separated list of variants.
+:--all:                    Selects all the experiments.
+:--axes [axes...]:         Selects all experiments with the variant axes from
+                           the space separated list of *axes*.
+:--run <r>:                Selects the single run given as
+                           ``<experiment_display_name>/<instance>``, where
+                           ``<experiment_display_name>`` is the name of the
+                           experiment as displayed on the command-line and
```
Comment (fabratu) [inline docs/command_line_reference.rst:98]: Same here.

File: docs/command_line_reference.rst [hunk 8]
```diff
@@ -41,68 +44,99 @@ command above).
...
+                           ``<instance>`` is the instance name as displayed on
+                           the command-line.
```
Comment (fabratu) [inline docs/command_line_reference.rst:100]: Same here.

File: docs/command_line_reference.rst [hunk 9]
```diff
@@ -41,68 +44,99 @@ command above).
...
-:list: lists all the instances found in the experiments.yml file.
-   Available instances are shown in green, unavailable instances in red.
-:process: caches information about instances.
-:run-transform: manually runs a transformation on instance files.
+The instance command can check the availability, and install instances. It
+supports the following flags:
+
+:install:         Installs missing instances. This can also trigger a download if an
+                  instance resource is configured accordingly. With the argument
+                  ``--overwrite``, this command will re-install all instances and
```
Comment (fabratu) [inline docs/command_line_reference.rst:118]: Same here (`re`)

File: docs/command_line_reference.rst [hunk 10]
```diff
@@ -41,68 +44,99 @@ command above).
...
+                  unavailable instances in red.
+
+:process:         Caches information about instances.
+
+:run-transform:   Manually runs the defined transformation on instance files.
+
+queue
+-----
+
+Triggers actions, or prints information concerning the configured experiment
```
Comment (fabratu) [inline docs/command_line_reference.rst:131]: I think the comma is not needed here.

**Review by fabratu — CHANGES_REQUESTED** [review]

File: docs/command_line_reference.rst (hunk 1 above)
Comment (c-bebop) [reply docs/command_line_reference.rst]: Done.

File: docs/command_line_reference.rst (hunk 3 above)
Comment (c-bebop) [reply docs/command_line_reference.rst]: Done.

File: docs/command_line_reference.rst (hunk 5 above)
Comment (c-bebop) [reply docs/command_line_reference.rst]: Done.

File: docs/command_line_reference.rst (hunk 7 above)
Comment (c-bebop) [reply docs/command_line_reference.rst]: Done.

File: docs/command_line_reference.rst (hunk 8 above)
Comment (c-bebop) [reply docs/command_line_reference.rst]: Done.

File: docs/command_line_reference.rst (hunk 4 above)
Comment (c-bebop) [reply docs/command_line_reference.rst]: Done.

File: docs/command_line_reference.rst (hunk 9 above)
Comment (c-bebop) [reply docs/command_line_reference.rst]: Done,

File: docs/command_line_reference.rst (hunk 10 above)
Comment (c-bebop) [reply docs/command_line_reference.rst]: Interesting point. I just learned a bit more about coordinate conjunctions :smile_cat:

File: docs/command_line_reference.rst (hunk 6 above)
Comment (c-bebop) [reply docs/command_line_reference.rst:86]: Renamed to *the scheduler*. Thanks for the hint!

**Review by fabratu — APPROVED** [review]

File: docs/command_line_reference.rst [hunk 11]
```diff
@@ -41,68 +44,99 @@ command above).
...
+:--delete-source: Deletes the source directory when purging.
+:--install:       Installs the build files.
+:--purge:         Deletes all related build files.
+:--recheckout:    Deletes the cloned git repository, reclones, regenerates,
+                  reconfigures, recompiles, and reinstalls it.
+:--recompile:     Recompiles and reinstall the build.
+:--reconfigure:   Reconfigures the build.
+:--regenerate:    Regenerates the build.
+:--reinstall:     Reinstalls the build.
+:--reregenerate:  Regenerates, reconfigures, recompiles, and reinstalls the build.
```
Comment (bernlu) [inline docs/command_line_reference.rst:59]: honestly I still do not understand what this command is supposed to do. If we regenerate something, then there has to be a generate step somewhere, but i do not see it?

File: docs/command_line_reference.rst [hunk 12]
```diff
@@ -41,68 +44,99 @@ command above).
...
+:--compile:       Compiles the build.
+:--configure:     Configures the build.
+:--delete-source: Deletes the source directory when purging.
+:--install:       Installs the build files.
+:--purge:         Deletes all related build files.
+:--recheckout:    Deletes the cloned git repository, reclones, regenerates,
+                  reconfigures, recompiles, and reinstalls it.
+:--recompile:     Recompiles and reinstall the build.
+:--reconfigure:   Reconfigures the build.
+:--regenerate:    Regenerates the build.
```
Comment (bernlu) [inline docs/command_line_reference.rst:57]: same comment as reregenerate

File: docs/command_line_reference.rst [hunk 13]
```diff
@@ -41,68 +44,99 @@ command above).
...
-:``--axes [axes...]``: selects all experiments with the variant axes from the space separated list of axes.
-:``--run <r>``: selects the single run given as ``<experiment_display_name>/<instance>``, where
-    ``<experiment_display_name>`` is the name of the experiment as displayed on the command-line and
-    ``<instance>`` is the instance name as displayed on the command-line.
-:``--experiment <e>``: selects the experiment named ``e``.
-:``--failed``: selects all the failed experiments.
-:``--instance <i>``: selects all experiments with the instance named ``i``.
-:``--instset <i>``: selects all experiments with the instance set named ``i``.
-:``--unfinished``: selects all the unfinished experiments.
-:``--revision <i>``: selects all experiments with the revision named ``r``.
```
Comment (bernlu) [inline docs/command_line_reference.rst:93]: `<i>` / `r` inconsistent

File: docs/command_line_reference.rst [hunk 14]
```diff
@@ -41,68 +44,99 @@ command above).
...
-It supports the following actions:
 
-:install: downloads all the missing instances if they are taken from a public repository.
-   With the argument ``--overwrite`` it will also download the available instances and
-   overwrite them.
-:list: lists all the instances found in the experiments.yml file.
-   Available instances are shown in green, unavailable instances in red.
-:process: caches information about instances.
-:run-transform: manually runs a transformation on instance files.
+The instance command can check the availability, and install instances. It
```
Comment (bernlu) [inline docs/command_line_reference.rst:113]: missing an `of` after availability?

File: docs/command_line_reference.rst [hunk 15]
```diff
@@ -41,68 +44,99 @@ command above).
...
+
+:install:         Installs missing instances. This can also trigger a download if an
+                  instance resource is configured accordingly. With the argument
+                  ``--overwrite``, this command will reinstall all instances and
+                  overwrite existing ones.
+
+:list:            Lists all defined instances. Available instances are shown in green,
+                  unavailable instances in red.
+
+:process:         Caches information about instances.
```
Comment (bernlu) [inline docs/command_line_reference.rst:124]: very unclear explanation. what is cached here?

**Review by bernlu — CHANGES_REQUESTED** [review]
In general the "Responsible to xy" phrase in each section sounds weird to me. Seems like the Oxford dictionary allows it, but to me it feels very unusual. Can we change this to "Responsible for" or some other word instead of responsible? `Instances` and `Queue` have better intro text imo.

File: docs/command_line_reference.rst (hunk 11 above)
Comment (c-bebop) [reply docs/command_line_reference.rst:59]: Read: https://simexpal.readthedocs.io/en/latest/builds.html#automated-builds
It's basically one phase before build that will be executed.

File: docs/command_line_reference.rst (hunk 12 above)
Comment (c-bebop) [reply docs/command_line_reference.rst:57]: See my comment on `--reregenerate`

File: docs/command_line_reference.rst (hunk 14 above)
Comment (c-bebop) [reply docs/command_line_reference.rst]: Done.

File: docs/command_line_reference.rst (hunk 15 above)
Comment (c-bebop) [reply docs/command_line_reference.rst]: Please look at new new change. I will create a new issue to actually remove this functionality. We do not only use NetworKit for this feature, which brings in another dependency, but is also very specific for graph files. I don't think that the value of this feature justifies the NetworKit dependency and maintenance of the feature.

File: docs/command_line_reference.rst (hunk 13 above)
Comment (c-bebop) [reply docs/command_line_reference.rst:93]: Thanks, fixed.

**Comment by c-bebop** [comment]
> In general the "Responsible to xy" phrase in each section sounds weird to me. Seems like the Oxford dictionary allows it, but to me it feels very unusual. Can we change this to "Responsible for" or some other word instead of responsible? `Instances` and `Queue` have better intro text imo.

Just changed this to '*Used to...*'.

File: docs/command_line_reference.rst (hunk 11 above)
Comment (bernlu) [reply docs/command_line_reference.rst:60]: okay, thanks :)

**Review by bernlu — APPROVED** [review]

File: docs/command_line_reference.rst (hunk 15 above)
Comment (bernlu) [reply docs/command_line_reference.rst]: I like the new explanation and I agree that this functionality is too specific.

**Pull request merged**